internal/config/yaml_reader.go

```diff
@@ -0,0 +1,56 @@
+package config
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+
+	"gopkg.in/yaml.v3"
+)
+
+const defaultConfigFile = "eph.yaml"
+
+var (
+	// Feature flag: if false, the function will default to using ./eph.yaml
+	AllowCustomConfigPath = false
+)
+
+// readYAMLFile reads a YAML file at the given path
+// and unmarshals it into a generic map[string]interface{}.
+func readYAMLFile(path string) (any, error) {
+	var filePath string
+	if AllowCustomConfigPath && len(path) > 0 && path != "" {
+		filePath = path
+	} else {
+		// Default to toplevel `eph.yaml`
+		cwd, err := os.Getwd()
+		if err != nil {
+			return nil, err
+		}
+		filePath = filepath.Join(cwd, defaultConfigFile)
+	}
+
+	// Read file into memory
+	data, err := os.ReadFile(filePath)
```

Comment:
```suggestion
	data, err := os.ReadFile(filePath) /* nose G305 user input for filepath is ok for yaml config */
```
I think we can safely ignore that rule here.  We aren't executing code that they are specifying, we are passing their config to apply changes to their environment.